cli/js/lib.deno.ns.d.ts

```diff
@@ -1863,6 +1863,147 @@ declare namespace Deno {
    * Requires `allow-run` permission. */
   export function run<T extends RunOptions = RunOptions>(opt: T): Process<T>;
 
+  /** The name of a "powerful feature" which needs permission.
+   *
+   * See: https://w3c.github.io/permissions/#permission-registry
+   *
+   * Note that the definition of `PermissionName` in the above spec is swapped
+   * out for a set of Deno permissions which are not web-compatible. */
+  export type PermissionName =
+    | "run"
+    | "read"
+    | "write"
+    | "net"
+    | "env"
+    | "plugin"
+    | "hrtime";
+
+  /** The current status of the permission.
+   *
+   * See: https://w3c.github.io/permissions/#status-of-a-permission */
+  export type PermissionState = "granted" | "denied" | "prompt";
+
+  export interface RunPermissionDescriptor {
+    name: "run";
+  }
+
+  export interface ReadPermissionDescriptor {
+    name: "read";
+    path?: string;
+  }
+
+  export interface WritePermissionDescriptor {
+    name: "write";
+    path?: string;
+  }
+
+  export interface NetPermissionDescriptor {
+    name: "net";
+    url?: string;
+  }
```

Comment:
One thing to consider is if `NetPermissionDescriptor` won't be (breakingly?) more structured in the future.

Ref #2705.

Reply:
It would cause breaking changes in a lot of places.  My opinion is stabilising the API doesn't mean that all permission descriptors are forever immutable.  They simply are a reflection of the permissions that Deno supports, in the way it supports them.  Leaving some descriptors as "unstable" while keeping everything else seems like it would be of little value as well.